### ISSUE TITLE
Can't update auto-scaling groups

### ISSUE THREAD

**Issue by smiller171** [ansible/ansible-modules-core#967]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.8.4
##### Environment:

Running on Ubuntu 14.04, managing AWS API
##### Summary:

Possibly related to replace_all_instances, in 1.8.4, an auto scaling group can not be updated once it exists. This is not an issue in 1.8.2

The code below works on the first run, but not on update
##### Steps To Reproduce:

``` yaml
- name: Create ASG
  ec2_asg:
    name: "automation-workbench-ASG"
    region: us-east-1
    load_balancers: "automation-workbench-ELB"
    availability_zones: "{{azs}}"
    launch_config_name: '{{new_config}}'
    min_size: "{{min}}"
    max_size: "{{max}}"
    desired_capacity: "{{desired}}"
    vpc_zone_identifier: "{{private_subnets}}"
    tags:
      - Name: "automation-workbench"
      - stack: "automation"
      - layer: "workbench"
    health_check_type: ELB
    health_check_period: 600
    replace_all_instances: yes
    replace_batch_size: 1
    state: present
  register: asg
```
##### Expected Results:

New instances are added to auto scaling group and old ones are removed

```
TASK: [create_scaling_layer | Create ASG] ************************************* 
changed: [localhost]
```
##### Actual Results:

```
TASK: [create_scaling_layer | Create ASG] ************************************* 
failed: [localhost] => {"failed": true}
msg: BotoServerError: 400 Bad Request
<ErrorResponse xmlns="http://autoscaling.amazonaws.com/doc/2011-01-01/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>Duplicate subnets specified: [b, 9, s, u, b, n, e, t, -, b, 9, 7, e, s, u, b, n, e, t, -, 2, b, 4, b, 5]</Message>
  </Error>
  <RequestId>ff721af1-ce49-11e4-a899-1bd0c4ec9657</RequestId>
</ErrorResponse>


FATAL: all hosts have already failed -- aborting
```

**Comment by jsmartin** [comment]
Which version of boto are you using?

**Comment by jsmartin** [comment]
Also, can you tell me what vpc_zone_identifier looks like?

**Comment by smiller171** [comment]
Boto version 2.20.1
```yaml
dev_private_subnets: 'subnet-bc6d9597,subnet-b829f7e1,subnet-42b40b35'
```

**Comment by smiller171** [comment]
Boto version 2.20.1

``` yaml
private_subnets: 'subnet-bc6d9597,subnet-b829f7e1,subnet-42b40b35'
```

**Comment by jsmartin** [comment]
Can you try with the latest version of boto ? 2.34.0 or higher?

**Comment by jsmartin** [comment]
I cannnot reproduce this on the devel branch of ansible -- with both boto 2.20.1 and 2.34.0.. It does happen on 1.8.4.. so I think this can be closed.

**Comment by smiller171** [comment]
Sounds like it's already been fixed and I just need to wait for the next release.

**Comment by abadger** [comment]
@smiller171 If you want to test to make sure that the issue will be fixed in 1.9.0, we've just created 1.9.0rc2.  Instructions for checking it out of git are here: https://groups.google.com/forum/#!topic/ansible-project/KuOK9GSO8ok

I've also got an unofficial tarball here: https://toshio.fedorapeople.org/ansible-1.9.0/  if you'd prefer to build and install it or some unofficial rpms at https://copr.fedoraproject.org/coprs/toshio/ansible-upstream/ if you're running on fedora or rhel/centos.
